src/constraint.rs

```diff
@@ -833,14 +855,135 @@ impl ConstraintValidator for ContainerLengthConstraint {
+}
+
+impl ConstraintValidator for ByteLengthConstraint {
+    fn validate(&self, value: &OwnedElement, type_store: &TypeStore) -> ValidationResult {
+        // Check for null value
+        if value.is_null() {
+            return Err(Violation::new(
+                "byte_length",
+                ViolationCode::TypeMismatched,
+                &format!("expected a clob/blob found {:?}", value),
+            ));
+        }
+
+        // get the size of given bytes
+        let size = match value.ion_type() {
+            IonType::Clob | IonType::Blob => value.as_bytes().unwrap().len(),
+            _ => {
+                // return Violation if value is not an clob/blob
+                return Err(Violation::new(
+                    "byte_length",
+                    ViolationCode::TypeMismatched,
+                    &format!("expected a clob/blob but found {}", value.ion_type()),
+                ));
+            }
+        };
+
+        // get isl length as a range
+        let length_range: &Range = self.length();
+
+        // return a Violation if the clob/blob size didn't follow byte_length constraint
+        if !length_range.contains(&(size as i64).into()).unwrap() {
+            return Err(Violation::new(
+                "byte_length",
+                ViolationCode::InvalidLength,
+                &format!("expected byte length {:?} found {}", length_range, size),
+            ));
+        }
+
+        Ok(())
+    }
+}
+
+/// Implements an `codepoint_length` constraint of Ion Schema
+/// [codepoint_length]: https://amzn.github.io/ion-schema/docs/spec.html#codepoint_length
+#[derive(Debug, Clone, PartialEq)]
+pub struct CodePointLengthConstraint {
+    length_range: Range,
+}
+
+impl CodePointLengthConstraint {
+    pub fn new(length_range: Range) -> Self {
+        Self { length_range }
+    }
+
+    pub fn length(&self) -> &Range {
+        &self.length_range
+    }
+}
+
+impl ConstraintValidator for CodePointLengthConstraint {
+    fn validate(&self, value: &OwnedElement, type_store: &TypeStore) -> ValidationResult {
+        // Check for null value
+        if value.is_null() {
+            return Err(Violation::new(
+                "codepoint_length",
+                ViolationCode::TypeMismatched,
+                &format!("expected a string/symbol found {:?}", value),
+            ));
+        }
+
+        // get the size of given string/symbol Unicode codepoints
+        let size = match value.ion_type() {
+            IonType::String => value.as_str().unwrap().chars().count(),
+            IonType::Symbol => value.as_sym().unwrap().text().unwrap().chars().count(),
+            _ => {
+                // return Violation if value is not string/symbol
+                return Err(Violation::new(
+                    "codepoint_length",
+                    ViolationCode::TypeMismatched,
+                    &format!("expected a string/symbol but found {}", value.ion_type()),
+                ));
+            }
+        };
```

Comment:
suggestion—you can simplify by using [`as_str()`](https://docs.rs/ion-rs/0.5.0/ion_rs/value/trait.Element.html#tymethod.as_str) for symbols and strings.
```suggestion
        // get the size of given string/symbol Unicode codepoints
        let size = match value.as_str() {
            Some(text) => text.chars().count(),
            _ => {
                // return Violation if value is not string/symbol
                return Err(Violation::new(
                    "codepoint_length",
                    ViolationCode::TypeMismatched,
                    &format!("expected a string/symbol but found {}", value.ion_type()),
                ));
            }
        };
```